### ISSUE TITLE
Ctrl D , Ctrl U not working when code folded

### ISSUE THREAD

**Issue by Shubhamsm** [VSCodeVim/Vim#5303]
**Describe the bug**

Navigating using Ctrl-D Ctrl-U not working when code is folded 
cursor stuck/freeze when it encounters a folded code and 

**Environment (please complete the following information):**

- Extension (VsCodeVim) version: v1.17.1
- VSCode version: VScodium v1.49.2
- OS: Kubuntu 20.04

**Comment by JW9506** [comment]
As an alternative, use `L`, `H` if you have set `"editor.cursorSurroundingLines": 3`, or use native vscode command `scrollLineDown`, `scrollPageDown`.

**Comment by BlameTroi** [comment]
VSCode v1.50.1, VsCodeVim v1.17.1 on Windows 10, I see the same behavior as @Shubhamsm and the problem also occurs with ctrl-f and ctrl-b.

**Comment by J-Fields** [comment]
This is more or less a duplicate of #5088, see [this comment](https://github.com/VSCodeVim/Vim/issues/5088#issuecomment-666509588)

**Comment by BlameTroi** [comment]
Thanks for the update. Good enough for me. Too bad VSCode isn't being friendly wrt folds. Carrying on.

**Comment by uloco** [comment]
Is there any way we can fix this at the current stage of vscode? Does anyone know if there has been improvements in the fold api?

**Comment by tobealive** [comment]
yeah, this drove me crazy. Avoided using code because of it. But I wrote an extension to solve it.

https://github.com/tobealive/german-scroll.code 👍

**Comment by prophetw** [comment]
> ctrl+d  half page scroll
> vscode scrollPageDown hole page scroll 
> How about use vscode scrollPageDown to handle ctrl+d 
```json
{
  "vim.normalModeKeyBindingsNonRecursive": [
    {
      "before": [
        "<C-d>"
      ],
      "commands": [
        "scrollPageDown"
      ]
    },
    {
      "before": [
        "<C-u>"
      ],
      "commands": [
        "scrollPageUp"
      ]
    }
  ],

}

```

**Comment by uloco** [comment]
@prophetw 

This unfortunately does not move your cursor but just the view which is not the same behavior.

**Comment by prophetw** [comment]
> @prophetw
> 
> This unfortunately does not move your cursor but just the view which is not the same behavior.

You're right!
So at last i do some tricks like this 


```json
    {
      "before": [
      "<C-d>"
      ],
      "after": [
        "1",
        "5",
        "g",
        "j",
        "z",
        "z",
      ]
    },
    {
      "before": [
        "<C-u>"
      ],
      "after": [
        "1",
        "5",
        "g",
        "k",
        "z",
        "z"
      ]
    },

```

**Comment by miranquil** [comment]
> @uloco You're right! So at last i do some tricks like this
> 
> ...

This is awesome!

**Comment by cpakken** [comment]
> @uloco You're right! So at last i do some tricks like this
> 
> ```json
>     {
>       "before": [
>       "<C-d>"
>       ],
>       "after": [
>         "1",
>         "5",
>         "g",
>         "j",
>         "z",
>         "z",
>       ]
>     },
>     {
>       "before": [
>         "<C-u>"
>       ],
>       "after": [
>         "1",
>         "5",
>         "g",
>         "k",
>         "z",
>         "z"
>       ]
>     },
> ```

This is actually better than the native implementation of <C-d> and <C-u>. I've always wanted to set the scroll distance a little less than half a page. Now I can! ❤

**Comment by OkkarMin** [comment]
> @uloco You're right! So at last i do some tricks like this
> 
> ```json
>     {
>       "before": [
>       "<C-d>"
>       ],
>       "after": [
>         "1",
>         "5",
>         "g",
>         "j",
>         "z",
>         "z",
>       ]
>     },
>     {
>       "before": [
>         "<C-u>"
>       ],
>       "after": [
>         "1",
>         "5",
>         "g",
>         "k",
>         "z",
>         "z"
>       ]
>     },
> ```

Its 2023 and this solution is simple and cleanest

**Comment by EnhaoSun** [comment]
Genious!

**Comment by huydq189** [comment]
for all who struggle with this issue like me
install these extensions:

- [https://github.com/tobealive/german-scroll.code](https://github.com/tobealive/german-scroll.code) (thanks @ttytm for a great extension)

- https://marketplace.visualstudio.com/items?itemName=kaiwood.center-editor-window

then add these to vscode setting

```
"vim.normalModeKeyBindingsNonRecursive": [
    {
      "before": ["<C-u>"],
      "commands": ["germanScroll.bertholdUp", "center-editor-window.center"]
    },
    {
      "before": ["<C-d>"],
      "commands": ["germanScroll.bertholdDown", "center-editor-window.center"]
    }]
```

**Comment by ooyay** [comment]
> @uloco You're right! So at last i do some tricks like this
> 
> ...

Without zz works better for me, tysm<3
```json


  "vim.normalModeKeyBindingsNonRecursive": [
    {
      "before": ["<C-d>"],
      "after": ["1", "5", "g", "j"]
    },
    {
      "before": ["<C-u>"],
      "after": ["1", "5", "g", "k"]
    }
  ],
```

**Comment by snobb** [comment]
Nice thanks everybody ended up doing this:
```
    {
      "before": [
        "<C-d>"
      ],
      "after": [
        "1", "5", "g", "j"
      ]
    },
    {
      "before": [
        "<C-u>"
      ],
      "after": [
        "1", "5", "g", "k"
      ]
    },
    {
      "before": [
        "<C-f>"
      ],
      "after": [
        "2", "5", "g", "j"
      ]
    },
    {
      "before": [
        "<C-b>"
      ],
      "after": [
        "2", "5", "g", "k"
      ]
    }
```

**Comment by LamNguyenz** [comment]
> [@uloco](https://github.com/uloco) You're right! So at last i do some tricks like this
> 
>     {
>       "before": [
>       "<C-d>"
>       ],
>       "after": [
>         "1",
>         "5",
>         "g",
>         "j",
>         "z",
>         "z",
>       ]
>     },
>     {
>       "before": [
>         "<C-u>"
>       ],
>       "after": [
>         "1",
>         "5",
>         "g",
>         "k",
>         "z",
>         "z"
>       ]
>     },

2025 and this is the best simplest and cleanest solution for me

**Comment by kettlemaster** [comment]
Edit: Nevermind, I went with the solution proposed by others in the end.

**Comment by snobb** [comment]
@kettlemaster Thanks, it's interesting. Indeed the `vim.foldfix` is not perfect when it comes to multiline actions. Speaking of folding - it's the biggest painpoint for me in the vscode(ium). Sadly it's not configurable and is less helpful than I'd hope comparing to the vanilla Vim (in addition to the issues you mentioned with the multiline behaviours, etc). I generally try to stick to the vanilla Vim, but sadly lately I must use some of the functionality (addons) required for my work that are not avialable in Vim (eg. Gitlab workspace that enables Duo suggestions that can only work in neovim, but not in the vanilla vim). Apart from behaviours, the information rendered for folding is not sufficent (eg. how many lines are folder, vim also shows a fragment of the first line within the fold, which works particularly well with Go unit test tables, etc).

I've posted a question in stack overflow on the subject, but I know now that sadly there isn't something that can be done with the folding.
https://stackoverflow.com/questions/78920079/is-there-a-way-to-customize-the-behaviour-of-code-folding-in-visual-studio-code

It's a shame the scrolling settings that you use do not move the cursor, so you need an extra keystroke to true it up. Thank you very much for sharing 🚀 - I'll try your settings and see if it works better than the fix that was proposed here previously.
